src/commands/scaling/addScaleRule/AddScaleRuleStep.ts

```diff
@@ -0,0 +1,78 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ScaleRule, Template } from "@azure/arm-appcontainers";
+import { AzureWizardExecuteStep, nonNullProp } from "@microsoft/vscode-azext-utils";
+import { Progress, window } from "vscode";
+import { ScaleRuleTypes } from "../../../constants";
+import { ext } from "../../../extensionVariables";
+import { localize } from "../../../utils/localize";
+import { updateContainerApp } from "../../updateContainerApp";
+import { IAddScaleRuleWizardContext } from "./IAddScaleRuleWizardContext";
+
+export class AddScaleRuleStep extends AzureWizardExecuteStep<IAddScaleRuleWizardContext> {
+    public priority: number = 100;
+
+    public async execute(context: IAddScaleRuleWizardContext, _progress: Progress<{ message?: string | undefined; increment?: number | undefined }>): Promise<void> {
+        const adding = localize('addingScaleRule', 'Adding {0} rule "{1}" to "{2}"...', context.ruleType, context.ruleName, context.containerApp.name);
+        const added = localize('addedScaleRule', 'Successfully added {0} rule "{1}" to "{2}".', context.ruleType, context.ruleName, context.containerApp.name);
+
+        const template: Template = context.containerApp.data.template || {};
+        template.scale = context.scale.data || {};
+        template.scale.rules = context.scaleRuleGroup.data || [];
+
+        const scaleRule: ScaleRule = this.buildRule(context);
+        this.integrateRule(context, template.scale.rules, scaleRule);
+
+        ext.outputChannel.appendLog(adding);
+        await updateContainerApp(context, context.containerApp, { template });
+        context.scaleRule = scaleRule;
```

Comment:
nit: I don't think that there's really any point in doing this. You're not really using it anywhere else, as far as I can tell.

Reply:
Yeah good catch, this had a specific use in a previous iteration, but now has gone the way of the appendix 😂

Reply:
Whoops